### PR TITLE
Changes close icon from img to svg to support currentColor

### LINKED ISSUE #244: Close icon should use currentColor
**I'm submitting a ...**
```
[ ] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request => Please do not submit support request here, post on Stackoverflow or Gitter
```

**Current behavior**

When the background color of the modal is changed to something dark (to match the OS preferences for dark mode e.g.) the close icon remains black due to the fact, that it uses `#000000` to fill the icon.

**Expected behavior**

The base64 encoded svg should be replaced with its corresponding svg code and use `currentColor` to fill the icon.
```xml
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" id="Layer_1" x="0px" y="0px" viewBox="0 0 512 512" style="enable-background:new 0 0 512 512;" xml:space="preserve" width="16px" height="16px">
<g>
    <path d="M505.943,6.058c-8.077-8.077-21.172-8.077-29.249,0L6.058,476.693c-8.077,8.077-8.077,21.172,0,29.249    C10.096,509.982,15.39,512,20.683,512c5.293,0,10.586-2.019,14.625-6.059L505.943,35.306    C514.019,27.23,514.019,14.135,505.943,6.058z" fill="currentColor"/>
</g>
<g>
    <path d="M505.942,476.694L35.306,6.059c-8.076-8.077-21.172-8.077-29.248,0c-8.077,8.076-8.077,21.171,0,29.248l470.636,470.636    c4.038,4.039,9.332,6.058,14.625,6.058c5.293,0,10.587-2.019,14.624-6.057C514.018,497.866,514.018,484.771,505.942,476.694z" fill="currentColor"/>
</g>
</svg>
```

And then maybe provide an additional (optional) property to color the icon separately from the modal text color if needed:
`<ngx-smart-modal ... [icon]="'red'">...</ngx-smart-modal>` which would add the provided color (or hexcode) to the surrounding `button` element.

**Reproduction of the problem**

https://plnkr.co/edit/j3kEpFufYkQV2PEXXj3S?p=preview

**What is the motivation / use case for changing the behavior?**

Dark UIs (e.g.) are fairly common nowadays and are pushed through OS Dark Modes. Therefore the close icon wouldn't be visible anymore if the UI changes to some dark theme.

### PR DESCRIPTION
**Pull request by marcreichel**
Adds closeIconColor property to change color of close icon if needed
Updates documentation accordingly
Updates contributor section in README

Resolves #244

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ngx-smart-modal-doc",
-  "version": "8.0.0",
+  "version": "8.1.0",
```
Comment (khylias) [inline package.json]: This package.json is for the demo and documentation apps.

File: src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts [hunk 2]
```diff
@@ -49,6 +59,7 @@ export class NgxSmartModalComponent implements OnInit, OnDestroy {
   @Input() public hideDelay: number = 500;
   @Input() public autostart: boolean = false;
   @Input() public target: string = '';
+  @Input() public closeIconColor: string = 'inherit';
```
Comment (khylias) [inline src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts]: I don't think this input is necessary. If the developer wants to change the color, he can do so by using CSS. This is not a functional need of the modal.

**Comment by marcreichel** [comment]
@LouisAugry How do I do this? Never done this before.

**Comment by LouisAugry** [comment]
> @LouisAugry How do I do this? Never done this before.

It may help you https://www.internalpointers.com/post/squash-commits-into-one-git

**Comment by marcreichel** [comment]
@LouisAugry Hmm, now I have a new commit with all changes but the old ones are still there 🤔

**Comment by marcreichel** [comment]
Maybe I should create a new PR from a new fork 😅

**Comment by maximelafarie** [comment]
@marcreichel for future PR no need to squash your commits, since we're squashing it at merge time 😉 (cf. @LouisAugry & @khylias)

Thank you for your work, I'll take a look at your new PR! 🖖

**Pull request closed without merging**